ci/scripts/PKGBUILD

```
@@ -50,6 +52,12 @@ source_dir="$ARROW_HOME"
 
 cpp_build_dir=build-${CARCH}-cpp
 
+# This should be "release" for real R package
+cpp_build_type="debug"
```

This seems like it should be derived from an environment variable near the top of this script:
```suggestion
cpp_build_type="${CPP_BUILD_TYPE:-debug}"
```

Then the value of `depends` can include boost iff we're building debug and this script can be used to build release by setting `CPP_BUILD_TYPE=release` or so